### PR TITLE
Strikelines chart shop created (almost_

### PR DESCRIPTION
**Pull request by MadWasp79**
First try for pull request

### CODE REVIEW

File: app/src/main/java/com/strikelines/app/utils/StyledBoldTextView.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.strikelines.app.utils
+
+import android.content.Context
+import android.graphics.Typeface
+import android.util.AttributeSet
+import android.widget.TextView
+import org.w3c.dom.Text
+
+class StyledBoldTextView: TextView{
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/utils/StyledBoldTextView.kt]: Seems to be too much for making text bold (it should be styled easier)

File: app/src/main/java/com/strikelines/app/utils/SingletonHolder.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package com.strikelines.app.utils
+
+
+/**
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/utils/SingletonHolder.kt]: Better use data structure with Application singleton

File: app/src/main/java/com/strikelines/app/utils/NetworkUtils.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        listener.onResult(result)
+    }
+
+    private fun getRequest(url: String): String {
+        val obj = URL(url)
+        val response = StringBuilder()
+        with(obj.openConnection() as HttpURLConnection) {
+            requestMethod = "GET"
+
+            println("URL : $url")
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/utils/NetworkUtils.kt]: Delete or replase with Log.d

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    private fun downloadFreeChart(downloadurl: String) {
+        //todo download chart
+    }
+
+    private fun getIntentContents(bundle: Bundle?) {
+        try {
+            chart = bundle?.getParcelable(CHART_BUNDLE_KEY)
+            Log.d("Details", chart.toString())
+        } catch (e: Exception) {
+            e.printStackTrace()
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Log.w(e.getMessage(), e)

File: app/src/main/java/com/strikelines/app/utils/NetworkUtils.kt [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import java.net.URL
+
+
+
+class GetRequestAsync(private val url: String, private val listener: OnRequestResultListener) : AsyncTask<Void, Void, String>() {
+
+    override fun doInBackground(vararg p0: Void?): String? {
+        try {
+            return getRequest(url)
+        } catch (e: NullPointerException) {
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/utils/NetworkUtils.kt]: ??? catch (e: Exception) would do the same

File: app/src/main/java/com/strikelines/app/utils/NetworkUtils.kt [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            e.printStackTrace()
+        } catch (e: MalformedURLException) {
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/utils/NetworkUtils.kt]: MalformedURLException is subclass of IOException

File: app/src/main/java/com/strikelines/app/domain/models/Chart.kt [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    val description: String,
+    val downloadurl: String,
+    val imageurl: String,
+    val latitude: String,
+    val longitude: String,
+    val name: String,
+    val price: String,
+    val region: String,
+    val weburl: String
+) : Parcelable {
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/domain/models/Chart.kt]: Avoid parcelable unless it is exactly necessary

File: app/src/main/java/com/strikelines/app/domain/models/Charts.kt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package com.strikelines.app.domain.models
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/domain/models/Charts.kt]: Is this class needed at all?

File: app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+                    Html.fromHtml(chart.description.clearGarbadge(), 0)
+                else Html.fromHtml(chart.description.clearGarbadge())
+        GlideApp.with(details_image)
+                .load(chart.imageurl).placeholder(R.drawable.img_placeholder).into(details_image)
+        if (chart.downloadurl.isEmpty()) {
+            get_chart_btn.setOnClickListener { startActivity(AndroidUtils.getIntentForBrowser(chart.weburl)) }
+            get_chart_btn.text = "GET CHART - $${chart.price}"
+        } else {
+            get_chart_btn.setOnClickListener { downloadFreeChart(chart.downloadurl) }
+            get_chart_btn.text = "ADD CHART - FREE"
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/ui/DetailedPurchaseChartScreen.kt]: Externalize strings to resources!!!

File: app/src/main/java/com/strikelines/app/ui/PurchaseSqliteDbFilesFragment.kt [hunk 10]
```diff
@@ -1,27 +1,106 @@
...
+    }
+
+    fun openBrowser(url: String) {
+        startActivity(AndroidUtils.getIntentForBrowser(url))
+    }
+
+
+    private val repoCallback =  object:RepoCallback{
+        override fun onLoadingComplete(status: String) {
+            Log.w("inFragment", status)
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/ui/PurchaseSqliteDbFilesFragment.kt]: Warning message?

File: app/src/main/java/com/strikelines/app/ui/PurchaseSqliteDbFilesFragment.kt [hunk 11]
```diff
@@ -1,27 +1,106 @@
...
+        }
+
+        override fun isResourcesLoading(status: Boolean) {
+            Log.w("inFragment loading", status.toString())
```
Comment (vshcherb) [inline app/src/main/java/com/strikelines/app/ui/PurchaseSqliteDbFilesFragment.kt]: Warning message?

File: app/src/main/java/com/strikelines/app/utils/StyledBoldTextView.kt (hunk 1 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/utils/StyledBoldTextView.kt]: min api should be 16 (here 14) to use xml font families. Another option is to set font programmatically. This option seems easier

File: app/src/main/java/com/strikelines/app/utils/SingletonHolder.kt (hunk 2 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/utils/SingletonHolder.kt]: done

File: app/src/main/java/com/strikelines/app/domain/models/Chart.kt (hunk 7 above)
Comment (MadWasp79) [reply app/src/main/java/com/strikelines/app/domain/models/Chart.kt]: removed

**Pull request merged**